workspaces/arborist/lib/node.js

```diff
@@ -1344,9 +1369,100 @@ class Node {
     this.edgesOut.set(edge.name, edge)
   }
 
-  addEdgeIn (edge) {
+  recalculateOutEdgesOverrides () {
+    // For each edge out propogate the new overrides through.
+    for (const [, edge] of this.edgesOut) {
+      edge.reload(true)
+      if (edge.to) {
+        edge.to.updateOverridesEdgeInAdded(edge.overrides)
+      }
+    }
+  }
+
+  findSpecificOverrideSet (first, second) {
```

Comment:
☝️ `findSpecificOverrideSet` is generic as it doesn't reply on the `this` value. It can be a. static method or stand alone helper function.